src/Libraries/CoreNodes/Data.cs

```diff
@@ -548,19 +548,31 @@ public class DataNodeDynamoType
             /// </summary>
             public string Name { get; private set; }
             /// <summary>
-            /// The parent of the DataType, if inheriting (to be removed in later updates)
```

Comment:
I find it a little strange that you have created a new type that wraps a .net type directly  instead of using the `ProtoCore.Type` (the vm type) 

https://github.com/DynamoDS/Dynamo/blob/master/src/Engine/ProtoCore/Lang/Type.cs#L12

Have you considered this - I imagine there might be lots of repeated information for rank etc that the VM already knows.

Reply:
Hi @mjkkirschner - no, I haven't considered at all actually! I was going to suggest that I can look into refactoring the code by using the ProtoCore.Type but I don't think I can extend it because it is a `struct`. There are 2 properties missing in the ProtoCore.Type that I have in the DataNodeDynamoType which are related to the hierarchical structure of the types that make life a lot easier when visualizing the data in the View. I will check in with @saintentropy  and @twastvedt  to discuss this further though, thank you for the prompt!!

Reply:
We've made everything private/internal that we could - we'll look at this in a follow-up.